### ISSUE TITLE
Flyout ShowAt cause WinRT transform error in C++ WinRT

### ISSUE THREAD

**Issue by HO-COOH** [microsoft/microsoft-ui-xaml#7989]
### Describe the bug

Calling `MenuFlyout.ShowAt()` at a newly constructed `Window` causes `WinRT transform error` when using C++/WinRT with WinUI3, but using the same code does not cause such exception in C#.

### Steps to reproduce the bug

1. Make a C++/WinRT project with WinUI3.
2. Add this in `MainWindow.xaml`
```xml
    <StackPanel Orientation="Horizontal" HorizontalAlignment="Center" VerticalAlignment="Center">
        <Button x:Name="myButton" Click="myButton_Click">Click Me</Button>
    </StackPanel>
```
3. Use this C++ code
`MainWIndow.xaml.h`
```cpp
    struct MainWindow : MainWindowT<MainWindow>
    {
        MainWindow();
        void myButton_Click(winrt::Windows::Foundation::IInspectable const& sender, winrt::Microsoft::UI::Xaml::RoutedEventArgs const& e);
        winrt::Microsoft::UI::Xaml::Window window{ nullptr };
    };
```

```cpp
void winrt::winui3Cpp::implementation::MainWindow::myButton_Click(winrt::Windows::Foundation::IInspectable const& sender, winrt::Microsoft::UI::Xaml::RoutedEventArgs const& e)
{
    window = winrt::Microsoft::UI::Xaml::Window{};
    winrt::Microsoft::UI::Xaml::Controls::Frame frame;
    window.Content(frame);

    winrt::Microsoft::UI::Xaml::Controls::MenuFlyout flyout;

    frame.Loaded(
        [flyout, this](auto, auto)
        {
            winrt::Microsoft::UI::Xaml::Controls::Primitives::FlyoutShowOptions options;
            options.ShowMode(winrt::Microsoft::UI::Xaml::Controls::Primitives::FlyoutShowMode::Transient);
            flyout.ShowAt(window.Content(), options); //exception here
        }
    );
}
```
4. Click the button, the exception
![image](https://user-images.githubusercontent.com/42881734/205695736-a0b7cfe4-8cbb-4530-9011-99ebdc9f7c86.png)

I wonder why it calls `CoreApplication::GetCurrentView()`, because it's supposed to be null in WinUI3.

Use the same code but in C#
```cs
        public MainWindow()
        {
            this.InitializeComponent();
        }

        Window window = null;
        private void myButton_Click(object sender, RoutedEventArgs e)
        {
            window = new Window();
            Frame frame= new Frame();
            window.Content= frame;
            
            MenuFlyout flyout = new MenuFlyout();
            frame.Loaded += (_, _) =>
            {
                FlyoutShowOptions options = new FlyoutShowOptions
                {
                    ShowMode = FlyoutShowMode.Transient
                };
                flyout.ShowAt(window.Content, options);
            };
        }
```
The exception does not occur.

### NuGet package version

WinUI 3 - Windows App SDK 1.2 : 1.2.221116.1

### Windows version

Windows 11 (22H2): Build 22621

**Comment by Scottj1s** [comment]
@HO-COOH Like a C++ app, a C# app calls SharedHelpers::TryGetCurrentCoreApplicationView(), which you can see if you enable native debugging.  In both cases, the call is intentional.

**Comment by HO-COOH** [comment]
![image](https://user-images.githubusercontent.com/42881734/205795473-e05f15d8-1c09-4237-b10f-48dd0376e486.png)
Yep, after enabling native debugging, the same exception occcurs in C#. 
![image](https://user-images.githubusercontent.com/42881734/205795718-c02adfa8-eb48-4dbc-9c70-3db024d7d25a.png)

**Comment by HO-COOH** [comment]
Okay, this exception happens not only on the newly constructed window like OP, but also happens on `flyout.ShowAt(myButton, ...);`, please fix!

**Comment by jarno9981** [comment]
@anyone any update on this

**Comment by jarno9981** [comment]
> @anyone any update on this

Update From My Site Got Working

**Comment by JesseCol** [comment]
Hi, I've verified this is fixed in the latest WinAppSDK 1.3 and 1.4 stable builds.  We do still have some outstanding places in the code where first-chance exceptions are still fired, but this specific one should be gone.  Please holler if you still see it.  

Also note that if a first-chance exception like this is too noisy for you, you can turn them off in the "Exception Settings" window.  The noisy ones in a WinUI 3 app are usually "Win32 Exceptions" / "WinRT Originate Error".
